### PR TITLE
Clean up bin/haraka

### LINKED ISSUE #2087: Cannot create plugin with subdirectory within command line
### system info

Haraka | Haraka.js — Version: 2.8.14
 --- | :--- 
Node | v6.10.0
OS | Darwin Anoshs-MacBook-Pro.local 16.7.0 Darwin Kernel Version 16.7.0: Thu Jun 15 17:36:27 PDT 2017; root:xnu-3789.70.16~2/RELEASE_X86_64 x86_64
openssl | OpenSSL 0.9.8zh 14 Jan 2016

### Expected behavior

`haraka -c . -p foo/bar` to produce:

 - `./plugins/foo/bar.js`
 - `./docs/plugins/foo/bar.md`

```
Plugin foo/bar created
Now edit javascript in:    /Users/anoshmalik/Workspace/HarakaTest/plugins/foo/bar.js
Add the plugin to config:  /Users/anoshmalik/Workspace/HarakaTest/config/plugins
And edit documentation in: /Users/anoshmalik/Workspace/HarakaTest/docs/plugins/foo/bar.md
```

### Observed behavior

`haraka -c . -p foo/bar` to doesn't produce:

 - `./plugins/foo/bar.js`
 - `./docs/plugins/foo/bar.md`

```
warning: Unable to create file: Error: ENOENT: no such file or directory, open '/Users/anoshmalik/Workspace/HarakaTest/plugins/foo/bar.js'
warning: Unable to create file: Error: ENOENT: no such file or directory, open '/Users/anoshmalik/Workspace/HarakaTest/docs/plugins/foo/bar.md'
Plugin foo/bar created
Now edit javascript in:    /Users/anoshmalik/Workspace/HarakaTest/plugins/foo/bar.js
Add the plugin to config:  /Users/anoshmalik/Workspace/HarakaTest/config/plugins
And edit documentation in: /Users/anoshmalik/Workspace/HarakaTest/docs/plugins/foo/bar.md
```

### Steps to reproduce

Run `haraka -c . -p foo/bar`.

### PR DESCRIPTION
**Pull request by KingNoosh**
Fixes #2087 

Changes proposed in this pull request:
- Remove /auth folders from creation
- Creates folders if doesn't exist

Checklist:
- [x] docs updated
- [x] tests updated
- [x] [Changes](https://github.com/haraka/Haraka/blob/master/Changes.md) updated

### CODE REVIEW

**Comment by msimerson** [comment]
I'm curious, why use the `mkdirpsync` module rather than `mkdirp.sync()`?  Not that I'm a huge fan of herd mentality, but one is super obscure and the other is extremely widely used and also includes an async version, which may be useful.

**Comment by KingNoosh** [comment]
That makes more sense actually, I'll change this when I get home.

**Comment by msimerson** [comment]
There is at least [one more place](https://github.com/haraka/Haraka/blob/master/plugins/queue/quarantine.js#L27-L34) we use mkdirp, and we've reinvented the wheel there. Once you've added `mkdirp` as a dependency, that plugin should likely also be updated.

File: plugins/queue/quarantine.js [hunk 1]
```diff
@@ -77,9 +68,12 @@ exports.get_base_dir = function () {
...
     const plugin = this;
     const tmp_dir = path.join(plugin.get_base_dir(), 'tmp');
-    mkdirs(tmp_dir, parseInt('0770', 8), function () {
+    mkdirp(tmp_dir, function (err) {
+        if (err) {
+            plugin.logerror(`Unable to create ${tmp_dir}`);
+        }
         plugin.loginfo(`created ${tmp_dir}`);
-        done();
+        return done();
```
Comment (msimerson) [inline plugins/queue/quarantine.js]: return here is redundant.

**Comment by msimerson** [comment]
thanks @KingNoosh

**Pull request merged**